api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java

```diff
@@ -32,12 +32,14 @@
 import org.apache.cloudstack.api.response.GuestOSResponse;
 import org.apache.cloudstack.api.response.ZoneResponse;
 import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.acl.RoleType;
 
 import java.net.MalformedURLException;
 
 @APICommand(name = GetUploadParamsForIsoCmd.APINAME,
         description = "upload an existing ISO into the CloudStack cloud.",
         responseObject = GetUploadParamsResponse.class, since = "4.13",
+        authorized = {RoleType.Admin, RoleType.ResourceAdmin, RoleType.DomainAdmin, RoleType.User},
```

Comment:
GetUploadParamsForVolumeCmd doesn't have this, but users without root admin role can still upload volumes. Do we know why this is required for this API?

Reply:
@nlgordon For GetUploadParamsForVolumeCmd, permissions are given as database entries in create-default-role-api-mappings.sql file. we can give permissions in two ways one is using annotations and second is using role permission tables

Reply:
to be consistent i would say it is best to do it via role permissions.  does the community version update that sql file?

Reply:
In main, the access is granted through command like we did.